src/map.js

```diff
@@ -66,99 +75,148 @@ export default class GpxMap {
+            states: [{
+                icon: 'fa-map fa-lg',
+                stateName: 'default',
+                title: 'Zoom to all tracks',
+                onClick: (_btn, map) => {
+                    centre();
+                },
+            }],
+        }).addTo(this.map);
+        this.viewAll.disable();
+
+        this.switchTheme(this.options.theme);
+        this.requestBrowserLocation();
+    }
 
-        this.switchTheme(this.options.theme)
-        this.requestBrowserLocation()
+    clearScroll () {
+        this.scrolled = false;
+        this.map.addEventListener("movestart", this.markScrolled);
     }
 
     switchTheme(themeName) {
-        if (this.mapTiles)
-            this.mapTiles.removeFrom(this.map)
+        if (this.mapTiles) this.mapTiles.removeFrom(this.map);
 
-        this.mapTiles = leaflet.tileLayer.provider(themeName)
-        this.mapTiles.addTo(this.map, {detectRetina: true})
+        this.mapTiles = leaflet.tileLayer.provider(themeName);
+        this.mapTiles.addTo(this.map, {detectRetina: true});
     }
 
     updateOptions(opts) {
         if (opts.theme !== this.options.theme) {
-            this.switchTheme(opts.theme)
+            this.switchTheme(opts.theme);
+        }
+        
+        if (opts.lineOptions.existing) {
+            this.tracks.forEach(t => {
+                t.setStyle({
+                    color: opts.lineOptions.color,
+                    weight: opts.lineOptions.weight,
+                    opacity: opts.lineOptions.opacity,
+                });
+
+                t.redraw();
+            });
         }
 
-        this.tracks.forEach(t => {
-            t.setStyle({
-                color: opts.lineOptions.color,
-                weight: opts.lineOptions.weight,
-                opacity: opts.lineOptions.opacity
-            })
-
-            t.redraw()
-        })
-
-        this.options = opts
+        this.options = opts;
     }
 
     // Try to pull geo location from browser and center the map
     requestBrowserLocation() {
-        navigator.geolocation.getCurrentPosition((pos) => {
-            this.map.panTo([pos.coords.latitude, pos.coords.longitude])
-        })
+        navigator.geolocation.getCurrentPosition(pos => {
+            if (!this.scrolled && this.tracks.length == 0) {
+                this.map.panTo([pos.coords.latitude, pos.coords.longitude], {
+                    noMoveStart: true,
+                });
+                // noMoveStart doesn't seem to have an effect, see Leaflet
+                // issue: https://github.com/Leaflet/Leaflet/issues/5396
+                this.clearScroll();
+            }
+        });
     }
 
     addTrack(track) {
-        let line = leaflet.polyline(track, this.options.lineOptions)
-        line.addTo(this.map)
+        this.viewAll.enable();
+        let lineOptions = Object.assign({}, this.options.lineOptions);
+        if (lineOptions.detectColors) {
+            if (/-(Hike|Walk)\.gpx/.test(track.filename))
```

Comment:
I would suggest making these case insensitive regexes. 

Out of curiosity, how did you pick the colors?

Reply:
I used case-sensitive names because Strava's bulk export titles them with leading capitals, and I did not want to auto-color any activities other than the bulk export ones. I chose the colours fairy arbitrarily (red, cyan, green); they could be made dynamic in the settings pane.

Reply:
I'd love the ability to make them customizable, but that can be a future PR 🙂